### PR TITLE
Added flags support to grpc_op.

### PR DESCRIPTION
**Pull request by dgquintas**
Which includes its propagation to grpc_ioreq and validation mechanisms for checking that only known its are set ot in).

Also added an internal flag (GRPC_WRITE_INTERNAL_COMPRESS) mask for its use in signaling compressed messages.

### CODE REVIEW

File: src/core/surface/call.c [hunk 1]
```diff
@@ -1211,30 +1225,43 @@ grpc_call_error grpc_call_start_batch(grpc_call *call, const grpc_op *ops,
     op = &ops[in];
     switch (op->op) {
       case GRPC_OP_SEND_INITIAL_METADATA:
+        /* Flag validation: currently allow no flags */
+        if (op->flags) return GRPC_CALL_ERROR_INVALID_FLAGS;
```
Comment (ctiller) [inline src/core/surface/call.c]: op->flags != 0

Comment (ctiller) [inline src/core/surface/call.c]: (and elsewhere)

**Comment by ctiller** [comment]
Minor nit, but LGTM.

**Comment by ctiller** [comment]
Looks like some wrappers will also need to be updated.

**Comment by jtattermusch** [comment]
Looks like this breaks the windows tests?
https://grpc-testing.appspot.com/job/gRPC%20jenkins/34/language=csharp,platform=windows/console

**Comment by dgquintas** [comment]
Green :)

**Comment by ctiller** [comment]
Can you make it mergable?

**Comment by dgquintas** [comment]
Done.

**Comment by dgquintas** [comment]
Green again.

**Pull request merged**